internal/controller/k0smotron.io/k0smotroncluster_statefulset.go

```diff
@@ -52,17 +52,17 @@ func (r *ClusterReconciler) findStatefulSetPod(ctx context.Context, statefulSet
 func (r *ClusterReconciler) generateStatefulSet(kmc *km.Cluster) (apps.StatefulSet, error) {
 
 	labels := labelsForCluster(kmc)
+	annotations := annotationsForCluster(kmc)
 
 	statefulSet := apps.StatefulSet{
 		TypeMeta: metav1.TypeMeta{
 			APIVersion: "apps/v1",
 			Kind:       "StatefulSet",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        kmc.GetStatefulSetName(),
-			Namespace:   kmc.Namespace,
-			Labels:      labels,
-			Annotations: annotationsForCluster(kmc),
+			Name:      kmc.GetStatefulSetName(),
+			Namespace: kmc.Namespace,
+			Labels:    labels,
```

Comment:
Small nitpick: we can keep setting annotations here and change the hash adding piece to:
```
statefulSet.Annotations[statefulSetAnnotation] = controller.ComputeHash(&statefulSet.Spec.Template, statefulSet.Status.CollisionCount)
```

Reply:
I thought about it but as I use the `annotations` variable also for the pod annotations, `statefulset.Annotations` and `pod.Annotations` point to the same reference so if I add a field in `statefulset.Annotations` then I also add it for the pod (I don't think the pod should contain that hash). That's why I opted for this approach. I also wanted to avoid calling `annotationsForCluster()` twice